aten/src/ATen/native/Copy.cpp

```diff
@@ -132,7 +132,8 @@ Tensor & copy_(Tensor & self, const Tensor & src, bool non_blocking) {
   }
 
   auto iter = TensorIterator();
-  iter.check_and_add_output(self);
+  iter.set_check_mem_overlap(true);
```

Comment:
I don't see a test for `copy_`.

`copy_` is a really fundamental op. Because of it, it would be great to make sure there are no performance changes due to this change. It would be good to run a before/after benchmark.

Reply:
addressed